src/core/utils.js

```diff
@@ -104,41 +92,54 @@ function markAsOffending(elem, msg, title) {
 }
 
 // STRING HELPERS
+/**
+ * @param {"conjunction"|"disjunction"} type
+ * @param {"long"|"narrow"} style
+ */
+function joinFactory(type, style = "long") {
+  const formatter = new Intl.ListFormat(docLang, { style, type });
+  /**
+   * @param {string[]} items
+   * @param {(value: string, index: number, array: string[]) => any} [mapper]
+   */
+  return (items, mapper) => {
```

Comment:
How about something like:
```ts
return (items: T[], mapper?: (item: T) => any), getter?: (item: T) => string) => {}
```

Reply:
Added.

Reply:
I meant:
```diff
-   return (items, mapper) => { 
+   return (items: T[], mapper?: (item: T) => any), getter?: (item: T) => string) => {
+   // above, but in JS (not TS)
```

Reply:
Let's ignore this for now. We can make it more generic next time we need it.

Reply:
Actually maybe we can?